mirrord-layer/src/pod_api.rs

```diff
@@ -331,12 +330,23 @@ async fn create_job_pod_agent(
 
     while let Some(status) = stream.try_next().await? {
         match status {
-            WatchEvent::Added(_) => break,
+            WatchEvent::Added(pod) | WatchEvent::Modified(pod) => {
+                let pod_status = &pod
+                    .status
+                    .and_then(|status| status.phase)
+                    .ok_or(LayerError::PodStatusNotFound(pod.metadata.name))?;
```

Comment:
We shouldn't err on this as this can be other pod theoretically (just do next)

Reply:
Not sure what you mean by do next, how do we handle the Option then?

Reply:
Continue if None, probably can be done using `if let Some(m) else continue`

Reply:
I took care of it (I think?)